lib/seed/facility_seeder.rb

```diff
@@ -123,17 +119,25 @@ def create_district_regions(state_results, facility_group_results)
     end
 
     def create_block_regions(district_region_results)
-      block_count = district_region_results.ids.size
+      # Eagerly fetch block names to avoid duplicates
```

Comment:
@rsanheim Is this reason accurate?

Reply:
We should be able to move the sampling in the loop. `Region` can support it now, although there might be a higher rate of blocks with same names I suppose.

Reply:
Closing the loop on this, the problem is not app code, but the bulk import. The bulk import bypasses app-level hooks, so two blocks with the same name dont' get assigned unique slugs, and end up clashing in the DB.